src/server/main.py

```diff
@@ -1,30 +1,345 @@
+        except jwt.InvalidTokenError as e:
+            return make_response(
+                json.dumps({"message": f"Token is invalid: {e}"}),
+                401,
+                {"WWW-Authenticate": 'Basic realm ="Token is invalid!"'},
+            )
+
+        try:
+            connection: Connection = getDatabaseConnection()
+            sql: str = "SELECT uuid, email FROM users WHERE uuid=?"
+            cursor: Cursor = connection.cursor()
+            cursor.execute(sql, (data["uuid"],))
+            user_information: list[tuple[str, str]] = cursor.fetchall()
+            connection.close()
+        except SqlError as e:
+            return make_response(json.dumps({"message": f"Database connection error: {e}"}), 500)
+
+        unique_id: str = ""
+
+        if not user_information:
+            return make_response(
+                json.dumps({"message": "User does not exist"}),
+                401,
+                {"WWW-Authenticate": 'Basic realm ="User does not exist!"'},
+            )
+
+        # Retrieve user uuid
+        unique_id = user_information[0][0]
+
+        # Returns the current logged-in users context to the routes
+        return fun(unique_id, *args, **kwargs)
+
+    return decorated
+
+
+def revoke_token(token: str) -> None:
+    """Revoking activated token."""
+    decoded_token: dict[str, Any] = jwt.decode(
+        token, app.config["SECRET_KEY"], algorithms=["HS256"]
+    )
+    expiry: datetime = datetime.fromtimestamp(decoded_token["exp"])
+
+    # Saving revoked token into the database
+    try:
+        connection: Connection = getDatabaseConnection()
+        connection.execute(
+            "INSERT INTO revoked_tokens (token, expiry) VALUES (?, ?)",
+            (
+                token,
+                expiry,
+            ),
+        )
+        connection.commit()
+        connection.close()
+    except SqlError as e:
+        raise SqlError(f"Database connection error: {str(e)}") from e
+
+
+def is_token_revoked(token: str) -> bool:
+    """Check if token is revoked."""
+    try:
+        connection: Connection = getDatabaseConnection()
+        result: Cursor = connection.execute(
+            "SELECT token FROM revoked_tokens WHERE token=? AND expiry > ?",
+            (
+                token,
+                datetime.utcnow(),
+            ),
+        )
+        is_revoked: bool = result.fetchone() is not None
+        connection.close()
+    except SqlError:
+        return True
+
+    return is_revoked
+
+
+@app.route("/register", methods=["POST"])
+def register() -> Response:
+    """Registration endpoint."""
+    new_user: dict[str, str] = request.json
+
+    unique_id: UUID = uuid.uuid4()
+    try:
+        email: str = new_user["email"]
+        password: str = new_user["password"]
+        repeated_password: str = new_user["repeatedPassword"]
+    except KeyError as e:
+        return make_response(json.dumps({"message": f"Invalid Json format: {e}"}), 202)
```

Comment:
This can be changed to i.e.:
```python
email: str = new_user.get("email", "")  # 2nd arg is the default value in case the key does not exist
...
if not email or not password or ...:
    return make_response(...)
```

It's a suggestion to avoid unnecessary `try except` here, you can leave it as that or change it.

Reply:
I've made suggested change.